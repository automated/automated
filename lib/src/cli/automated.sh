COMMAND="$(pwd)/node_modules/@automated/automated/lib/dist/cli/runner.js $@"
echo x2
echo $AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED
node $COMMAND
