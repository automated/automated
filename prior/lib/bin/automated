#!/bin/bash

export NODE_ENV="automated"

### Get cli arguments
# maintain arg quotes - https://stackoverflow.com/a/24179878/537998
function requote() {
  local res=""
  for x in "${@}" ; do
      # try to figure out if quoting was required for the $x:
      grep -q "[[:space:]]" <<< "$x" && res="${res} '${x}'" || res="${res} ${x}"
  done
  # remove first space and print:
  sed -e 's/^ //' <<< "${res}"
}
cliArgs=$(requote "${@}")
###

pwd=`pwd`;
pluginDirSearch='/node_modules/@automated/[plugin-]*'

declare -a lifecycles=(
  collection-will-start
  collection-did-start
  collection-running
  test-will-start
  test-did-start
  test-running
  test-will-end
  collection-will-end
)

for lifecycle in "${lifecycles[@]}"
do
  for pluginDir in `ls -d $pwd$pluginDirSearch`
  do
    # look for something like `jest[`
    pluginWithBraket=$(echo $pluginDir | pcregrep -o1 'plugin-(.*)')\\[
    script=$pluginDir/$lifecycle
    if [[ -e $script ]]; then
      if [[ $cliArgs == *$pluginWithBraket* ]]; then
        # run with args
        pluginArgs=$(echo $cliArgs | pcregrep -o1 "$pluginWithBraket(.*)\]")
        cmd=$(echo $script $pluginArgs)
        # echo $cmd
        $cmd
      else
        # run without args
        cmd=$(echo $script)
        # echo $cmd
        $cmd
      fi
    fi
  done
done
