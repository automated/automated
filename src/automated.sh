COMMAND="$(pwd)/node_modules/@automated/automated/dist/cli/runner.js $@"
node $COMMAND
