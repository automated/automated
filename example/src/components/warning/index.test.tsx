/*
No props example
*/

import { runner } from '@automated/automated';
import Component from './';

runner({
  filename: __filename,
  process,
  Component,
});
