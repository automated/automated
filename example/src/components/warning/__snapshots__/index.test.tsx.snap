// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/src/components/warning/index.test.tsx default 1`] = `
.emotion-0 {
  padding: 100px;
  border-radius: 10px;
  background: red;
  color: white;
  font-weight: 800;
  font-size: 20px;
}

<div
  className="emotion-0"
>
  Warning!
</div>
`;
